Application crashes if user click on widget after the account has been deleted.

Reproduce: Add a widget for an account. Delete the account but keep the widget. Click on the widget and the application crashes.

```java
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.liato.bankdroid/com.liato.bankdroid.TransactionsActivity}: java.lang.NullPointerException
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2237)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2286)
       at android.app.ActivityThread.access$800(ActivityThread.java:144)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1246)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:212)
       at android.app.ActivityThread.main(ActivityThread.java:5135)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:877)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at com.liato.bankdroid.TransactionsActivity.onCreate(TransactionsActivity.java:55)
       at android.app.Activity.performCreate(Activity.java:5231)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2201)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2286)
       at android.app.ActivityThread.access$800(ActivityThread.java:144)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1246)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:212)
       at android.app.ActivityThread.main(ActivityThread.java:5135)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:877)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

```java
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.liato.bankdroid/com.liato.bankdroid.TransactionsActivity}: android.database.CursorIndexOutOfBoundsException: Index 0 requested, with a size of 0
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
       at android.app.ActivityThread.access$800(ActivityThread.java:144)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5221)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
Caused by: android.database.CursorIndexOutOfBoundsException: Index 0 requested, with a size of 0
       at android.database.AbstractCursor.checkPosition(AbstractCursor.java:426)
       at android.database.AbstractWindowedCursor.checkPosition(AbstractWindowedCursor.java:136)
       at android.database.AbstractWindowedCursor.getInt(AbstractWindowedCursor.java:68)
       at com.liato.bankdroid.banking.BankFactory.bankFromDb(BankFactory.java:50)
       at com.liato.bankdroid.TransactionsActivity.onCreate(TransactionsActivity.java:47)
       at android.app.Activity.performCreate(Activity.java:5933)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
       at android.app.ActivityThread.access$800(ActivityThread.java:144)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5221)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
```

--- a/app/src/main/java/com/liato/bankdroid/BankdroidApplication.java
+++ b/app/src/main/java/com/liato/bankdroid/BankdroidApplication.java
@@ -1,6 +1,7 @@
 package com.liato.bankdroid;
 
 import android.app.Application;
+import android.widget.Toast;
 
 import com.crashlytics.android.Crashlytics;
 import com.liato.bankdroid.banking.Bank;
@@ -15,6 +16,8 @@ public class BankdroidApplication extends Application {
 
     public static final String LOG_KEY_LOCALE = "locale";
     public static final String LOG_KEY_BANKS = "banks";
+
+    private String message = "";
 
     @Override
     public void onCreate() {
@@ -33,6 +36,18 @@ public class BankdroidApplication extends Application {
                         .append(",");
             }
             Crashlytics.setString(LOG_KEY_BANKS, bankStringBuilder.substring(0, bankStringBuilder.length() - 1));
+        }
+    }
+
+    public void setApplicationMessage(String messageText) {
+        message = messageText == null ? "" : messageText;
+    }
+
+    public void showAndDeleteApplicationMessage() {
+        if (!message.isEmpty()) {
+            Toast toast = Toast.makeText(this, message, Toast.LENGTH_LONG);
+            message = "";
+            toast.show();
         }
     }
 }

--- a/app/src/main/java/com/liato/bankdroid/MainActivity.java
+++ b/app/src/main/java/com/liato/bankdroid/MainActivity.java
@@ -118,7 +118,7 @@ public class MainActivity extends LockableActivity {
     			startActivity(intent);
 		    }
 		}
-        ((BankdroidApplication) getApplication()).showAndDeleteApplicationMessage(this);
+        ((BankdroidApplication) getApplication()).showAndDeleteApplicationMessage();
 	}
 
 	

--- a/app/src/main/java/com/liato/bankdroid/MainActivity.java
+++ b/app/src/main/java/com/liato/bankdroid/MainActivity.java
@@ -118,6 +118,7 @@ public class MainActivity extends LockableActivity {
     			startActivity(intent);
 		    }
 		}
+        ((BankdroidApplication) getApplication()).showAndDeleteApplicationMessage(this);
 	}
 
 	

--- a/app/src/main/java/com/liato/bankdroid/TransactionsActivity.java
+++ b/app/src/main/java/com/liato/bankdroid/TransactionsActivity.java
@@ -17,6 +17,7 @@
 package com.liato.bankdroid;
 
 import android.content.Context;
+import android.content.Intent;
 import android.os.Bundle;
 import android.view.LayoutInflater;
 import android.view.View;
@@ -45,7 +46,15 @@ public class TransactionsActivity extends LockableActivity {
         setContentView(R.layout.transactions);
         Bundle extras = getIntent().getExtras();
         Bank bank = BankFactory.bankFromDb(extras.getLong("bank"), this, false);
+        if(bank == null) {
+            redirectToMain(getString(R.string.error_bank_not_found));
+            return;
+        }
         Account account = BankFactory.accountFromDb(this, extras.getLong("bank") + "_" + extras.getString("account"), true);
+        if(account == null) {
+            redirectToMain(getString(R.string.error_account_not_found));
+            return;
+        }
         TextView viewBankName = (TextView) findViewById(R.id.txtListitemAccountsGroupAccountname);
         TextView viewAccountName = (TextView) findViewById(R.id.txtListitemAccountsGroupBankname);
         TextView viewAccountBalance = (TextView) findViewById(R.id.txtListitemAccountsGroupTotal);
@@ -172,6 +181,12 @@ public class TransactionsActivity extends LockableActivity {
             return 1;
         }
 
+    }
+
+    private void redirectToMain(String errorMessage) {
+        final Intent intent = new Intent(this, MainActivity.class);
+        ((BankdroidApplication) getApplicationContext()).setApplicationMessage(getString(R.string.error_bank_not_found));
+        startActivity(intent);
     }
 
 }

--- a/app/src/main/java/com/liato/bankdroid/banking/BankFactory.java
+++ b/app/src/main/java/com/liato/bankdroid/banking/BankFactory.java
@@ -44,7 +44,7 @@ public class BankFactory {
 		DBAdapter db = new DBAdapter(context);
 		Cursor c = db.getBank(id);
 
-		if (c != null) {
+		if (c != null && c.getCount() > 0) {
 			try {
 				bank = fromBanktypeId(c.getInt(c.getColumnIndex("banktype")), context);
 				String password = "";
@@ -82,7 +82,7 @@ public class BankFactory {
 		ArrayList<Bank> banks = new ArrayList<Bank>();
 		DBAdapter db = new DBAdapter(context);
 		Cursor c = db.fetchBanks();
-		if (c == null) {
+		if (c == null || c.getCount() == 0) {
 			return banks;
 		}
 		while (!c.isLast() && !c.isAfterLast()) {
@@ -162,7 +162,7 @@ public class BankFactory {
 		ArrayList<Account> accounts = new ArrayList<Account>();
 		DBAdapter db = new DBAdapter(context);
 		Cursor c = db.fetchAccounts(bankId);
-		if (c == null) {
+		if (c == null || c.getCount() == 0) {
 			return accounts;
 		}
 		while (!c.isLast() && !c.isAfterLast()) {

--- a/app/src/main/res/values-sv/strings.xml
+++ b/app/src/main/res/values-sv/strings.xml
@@ -247,4 +247,8 @@
     <!-- Color Picker -->
     <string name="dialog_color_picker">Färgväljare</string>
     <string name="press_color_to_apply">Tryck på färgen för att välja</string>
+
+    <!-- Error messages -->
+    <string name="error_bank_not_found">Den valda banken kunde inte hittas.</string>
+    <string name="error_account_not_found">Det valda kontot kunde inte hittas.</string>
 </resources>

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -302,4 +302,8 @@
     <!-- Color Picker -->
     <string name="dialog_color_picker">Color Picker</string>
     <string name="press_color_to_apply">Press on Color to apply</string>    
+
+    <!-- Error messages -->
+    <string name="error_bank_not_found">The chosen bank could not be found.</string>
+    <string name="error_account_not_found">The chosen account could not be found.</string>
 </resources>

